src/components/ConfirmProvider.tsx

```diff
@@ -0,0 +1,79 @@
+/* eslint-disable @typescript-eslint/no-empty-function */
+import { FormattedMessage } from 'react-intl';
+import { Box, Button } from '@material-ui/core';
+import React, { useContext, useState } from 'react';
+
+import ZetkinDialog from './ZetkinDialog';
+
+interface ConfirmProps {
+    actionText: string;
+    title: string;
+    onConfirm: <T>(t?: T) => void;
+}
+
+interface ContextProps {
+    confirmProps: ConfirmProps;
+    open: boolean;
+    reset: () => void;
+    setConfirmProps: (newProps: ConfirmProps) => void;
+    setOpen: (newValue: boolean) => void;
+}
+
+const defaultValue = {
+    confirmProps: { actionText: '', onConfirm: () => {} , title: '' },
+    open: false,
+    reset: () => null,
+    setConfirmProps: () => null,
+    setOpen: () => null,
+};
+
+export const ConfirmContext = React.createContext<ContextProps>(defaultValue);
+
+const ConfirmModal = () => {
+    const confirm = useContext(ConfirmContext);
+
+    const onClickConfirm = () => {
+        confirm.confirmProps.onConfirm();
+        clear();
+    };
+
+    const clear = () => {
+        confirm.setOpen(false);
+        confirm.reset();
+    };
+
+    return (
+        <ZetkinDialog
+            onClose={ clear }
+            open={ confirm.open }
+            title={ confirm.confirmProps.title }>
+            <Box>
+                <p>{ confirm.confirmProps.actionText }</p>
+                <Box display="flex" justifyContent="flex-end" py={ 2 }>
+                    <Box mx={ 2 }>
+                        <Button color="primary" onClick={ onClickConfirm } variant="contained">
+                            <FormattedMessage id="misc.components.confirm.buttons.submit" />
+                        </Button>
+                    </Box>
+                    <Button color="default" onClick={ () => confirm.setOpen(false) } variant="contained">
+                        <FormattedMessage id="misc.components.confirm.buttons.cancel" />
+                    </Button>
+                </Box>
```

Comment:
We have a component for this exact thing `SubmitCancelButtons.tsx` in `forms`

Reply:
The submit, rather than being called with an onclick, is called by wrapping the buttons in a `form` and using the `onSubmit` handler with `preventDefault`